lib/arch_obj_models/csr.rb

```
@@ -219,6 +226,32 @@ def max_length(cfg_arch)
       else
         64
       end
+    when "XLEN"
+      if cfg_arch.possible_extensions.map(&:name).include?("M")
+        cfg_arch.mxlen || 64
+      elsif cfg_arch.possible_extensions.map(&:name).include?("S")
+        if cfg_arch.param_values.key?("SXLEN")
+          if cfg_arch.param_values["SXLEN"] == 3264
+            64
+          else
+            cfg_arch.param_values["SXLEN"]
+          end
+        else
+          64
+        end
+      elsif cfg_arch.possible_extensions.map(&:name).include?("H")
+        if cfg_arch.param_values.key?("VSXLEN")
+          if cfg_arch.param_values["VSXLEN"] == 3264
+            64
+          else
+            cfg_arch.param_values["VSXLEN"]
+          end
+        else
+          64
+        end
+      else
+        64
```

I don't think we can ever get here, or if you can I'm not sure what it means ;) So let's `raise "unexpected"`